src/Security/src/Authorization.Certificate/ApplicationInstanceCertificate.cs

```diff
@@ -0,0 +1,62 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information.
+
+using System.Diagnostics.CodeAnalysis;
+using System.Text.RegularExpressions;
+
+namespace Steeltoe.Security.Authorization.Certificate;
+
+internal sealed class ApplicationInstanceCertificate
+{
+    // This pattern is found on certificates issued by Diego
+    private const string CloudFoundryInstanceCertificateSubjectRegex =
+        @"^CN=(?<instance>[0-9a-f-]+),\sOU=organization:(?<org>[0-9a-f-]+)\s\+\sOU=space:(?<space>[0-9a-f-]+)\s\+\sOU=app:(?<app>[0-9a-f-]+)$";
+
+    // This pattern is found on certificates created in Steeltoe
+    private const string SteeltoeInstanceCertificateSubjectRegex =
+        @"^CN=(?<instance>[0-9a-f-]+),\sOU=app:(?<app>[0-9a-f-]+)\s\+\sOU=space:(?<space>[0-9a-f-]+)\s\+\sOU=organization:(?<org>[0-9a-f-]+)$";
+
+    public string OrganizationId { get; private set; }
+
+    public string SpaceId { get; private set; }
+
+    public string ApplicationId { get; private set; }
+
+    public string InstanceId { get; private set; }
+
+    private ApplicationInstanceCertificate(string organizationId, string spaceId, string applicationId, string instanceId)
+    {
+        OrganizationId = organizationId;
+        SpaceId = spaceId;
+        ApplicationId = applicationId;
+        InstanceId = instanceId;
+    }
+
+    public static bool TryParse(X509Certificate2 certificate, [MaybeNullWhen(false)] out ApplicationInstanceCertificate outInstanceCertificate, ILogger logger)
```

Comment:
This signature doesn't match the "try" pattern, for example:
```c#
public static bool TryParse(ReadOnlySpan<char> ipSpan, [NotNullWhen(true)] out IPAddress? address)
```

Reply:
What's the point of logging in this method? All of the information logged is known by the caller.
Otherwise, this method should take an `ILogger<ApplicationInstanceCertificate>`.

Reply:
Both great points, it looks like I had done next to nothing with this class besides rename it

Reply:
The parameter `outInstanceCertificate` should be renamed without the `out` prefix.